src/TSTransformer/nodes/expressions/transformTemplateExpression.ts

```diff
@@ -1,29 +1,26 @@
 import luau from "@roblox-ts/luau-ast";
-import { errors } from "Shared/diagnostics";
-import { assert } from "Shared/util/assert";
 import { TransformState } from "TSTransformer";
-import { DiagnosticService } from "TSTransformer/classes/DiagnosticService";
 import { createStringFromLiteral } from "TSTransformer/util/createStringFromLiteral";
 import { ensureTransformOrder } from "TSTransformer/util/ensureTransformOrder";
-import { binaryExpressionChain } from "TSTransformer/util/expressionChain";
-import {
-	isDefinitelyType,
-	isLuaTupleType,
-	isPossiblyType,
-	isStringType,
-	isUndefinedType,
-} from "TSTransformer/util/types";
 import ts from "typescript";
 
-export function transformTemplateExpression(state: TransformState, node: ts.TemplateExpression) {
-	// if there are zero templateSpans, this must be a ts.NoSubstitutionTemplateLiteral
-	// and will be handled in transformStringLiteral
-	assert(node.templateSpans.length > 0);
+function createInterpolatedStringPart(node: ts.TemplateLiteralToken | ts.StringLiteral) {
+	return luau.create(luau.SyntaxKind.InterpolatedStringPart, { text: createStringFromLiteral(node) });
+}
```

Comment:
Maybe this should be `transformInterpolatedStringPart` since it converts the ts AST node to a luau AST node.

`createX` is _usually_ reserved for creating Luau nodes from anything that isn't a TS node.

Reply:
+ maybe it should live in it's own file, similar to previous comment